libmamba/src/core/url.cpp

```diff
@@ -16,8 +16,8 @@ namespace mamba
     {
         // usernames on anaconda.org can have a underscore, which influences the
         // first two characters
-        const static std::regex token_re("/t/([a-zA-Z0-9-_]{0,2}[a-zA-Z0-9-]*)");
-        const static std::regex http_basicauth_re("://([^\\s]+):([^\\s]+)@");
+        static constexpr auto token_pattern("/t/([a-zA-Z0-9-_]{0,2}[a-zA-Z0-9-]*)");
```

Comment:
I think the idea was that we don't have to recompile the regex every time we use it.

Reply:
Indeed, I realized that when writing this but had to confirm that the lifetime of the compiled regex was the main issue I was seeing, seems I was right.  I have been thinking about a better solution to keep the regex built once, hopefully it will be fixed in next commit.

Reply:
I setup a better fix as part of my cleanup/rebasing.